src/tree.jl

```diff
@@ -16,30 +16,30 @@ export Node,
 #     Any missing links (e.g., "c" does not have a sibling) link back to itself (c.sibling == c)
 # With this organization, no arrays need to be allocated.
 
-type Node{T}
+mutable struct Node{T}
     data::T
     parent::Node{T}
     child::Node{T}
     sibling::Node{T}
     
     # Constructor for the root of the tree
-    function (::Type{Node{T}}){T}(data::T)
+    function (::Type{Node{T}})(data::T) where T
```

Comment:
These can change to the newer and much easier syntax, `function Node{T})(data) where T`.